src/web/rustdoc.rs

```diff
@@ -795,6 +798,29 @@ impl Handler for SharedResourceHandler {
     }
 }
 
+/// Serves shared web resources used by rustdoc-generated documentation.
+///
+/// Rustdoc has certain JS, CSS, font and image files that are required for all
+/// documentation it generates, and these don't change often. We make one copy
+/// of these per rustdoc release and serve them from a common location.
+///
+/// This handler considers the whole path, and looks for a file at that path in
+/// the Storage.
+pub struct SharedResourceHandler;
+
+impl Handler for SharedResourceHandler {
+    fn handle(&self, req: &mut Request) -> IronResult<Response> {
+        let storage = extension!(req, Storage);
+        let config = extension!(req, Config);
+
+        let storage_path = format!("/{}", req.url.path().join("/"));
```

Comment:
This seems like it might be vulnerable to directory traversal issues? `https://docs.rs/-/rustdoc.static/../../sources/regex/1.6.0.zip`.

Reply:
(Maybe it's handled by the router already, but the static file handler has protection against it so that would need checking).

Reply:
we should decouple the given path in the URL from the path in the bucket here. 

So: 
1. strip the prefix elements (`/-/rustdoc.static/`) 
2. apply the bucket prefix
3. use that to request the file

Traversal could be checked similar to our `static_handler`, while it's definitely less risk since it would only allow downloading files from the bucket, which is private, but doesn't contain any risky files. 

Alternative: 
similar to the current `SharedResourceHandler`, just use the last element, not allowing assets in subdirectories.